### PR TITLE
Add codespell support (config, workflow to detect/not fix) and make it fix few typos

### PR DESCRIPTION
**Pull request by yarikoptic**
I think I have used btrfs-progs directly or not (via btrbk) for years... finally got a motivation/trigger to contribute ;)

More about codespell: https://github.com/codespell-project/codespell .

I personally introduced it to dozens if not hundreds of projects already and so far only positive feedback.

CI workflow has 'permissions' set only to 'read' so also should be safe.

### CODE REVIEW

**Comment by yarikoptic** [comment]
there was also this one which I later ignored since ignored all the tests, but which looks like a legit typo which might be masking desired test error/fail?

```
❯ codespell -C 4 ./tests/fsck-tests/052-init-csum-tree/test.sh
: run_check $SUDO_HELPER fallocate -p -o 4096 -l 4096 "$TEST_MNT/regular_with_holes"
: 
: # And the most complex one, preallocated, written, then hole
: run_check $SUDO_HELPER fallocate  -l 8192 "$TEST_MNT/complex"
> run_check $SUDO_HELPER dd if=/dev/urandom of="$TEST_MNT/compex" \
: 	bs=4k count=1 conv=notrunc status=noxfer > /dev/null 2>&1
: sync
: run_check $SUDO_HELPER fallocate -p -l 4096 "$TEST_MNT/regular_with_holes"
: 
./tests/fsck-tests/052-init-csum-tree/test.sh:46: compex ==> complex
```

see - that there is `compex` not `complex` ... Should I add that fix here too?

**Comment by kdave** [comment]
Right, `compex` is referring to a file that's used in previous comment so it's a typo and a bug in the test.

**Comment by kdave** [comment]
Thanks, the spellchecking is definitelly a good thing, the typos sneak in, you could find older commits that fix many of them in documentation, yet you found a lot more.

I've posted review comments, probably more exceptions to be added. The `compex` fix should be a standalone patch as it has functional consequences.

**Comment by yarikoptic** [comment]
> I've posted review comments, probably more exceptions to be added. 

Sorry, I don't see them -- may be review was not finalized/submitted?

> The `compex` fix should be a standalone patch as it has functional consequences.

- https://github.com/kdave/btrfs-progs/pull/846

**Comment by kdave** [comment]
Yeah sorry, I'm not yet familiar with the GH reviews, I'll publish them right away.

**Review by kdave — CHANGES_REQUESTED** [review]

File: Documentation/Kernel-by-version.rst [hunk 1]
```diff
@@ -678,7 +678,7 @@ Other:
 - code refactoring, file splits
 - preparatory work for more checksums
 - tree checker to verify lengths of various items
-- delayed iput happens at unlink time, not in cleaner thread
+- delayed input happens at unlink time, not in cleaner thread
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply Documentation/Kernel-by-version.rst:681]: Would you prefer to slip iput globally or just eg as "delayed iput" only ?

[the remark this answers is not in the document]
Comment (adam900710) [reply Documentation/Kernel-by-version.rst:681]: Please skip `iput` globally. There are also non-delayed iput, and it's pretty a common term used inside kernel too.

**Comment by yarikoptic** [comment]
- accounted for above review comments
- rebased
- used newer codespell (found few more typos)
- force-pushed

File: check/main.c [hunk 2]
```diff
@@ -10016,7 +10016,7 @@ static int check_global_roots_uptodate(void)
 	for (n = rb_first(&gfs_info->global_roots_tree); n; n = rb_next(n)) {
 		root = rb_entry(n, struct btrfs_root, rb_node);
 		if (!extent_buffer_uptodate(root->node)) {
-			error("chritical: global root [%llu %llu] not uptodate, unable to check the file system",
+			error("chritical: global root [%llu %llu] not up-to-date, unable to check the file system",
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply check/main.c:10020]: brr... I took it the other way around... fixing now...

**Comment by yarikoptic** [comment]
CI needs a restart or might be event better

- https://github.com/kdave/btrfs-progs/pull/848

merge/rebase

**Review by kdave — APPROVED** [review]

**Comment by kdave** [comment]
Merged to devel. I'll update the subject lines or changelogs a bit, we're using the prefixes or want to put some description.

**Comment by kdave** [comment]
After the apt-cache fix everthing works, now we wait for the typos to come.

**Pull request closed without merging**